db/user.go

```diff
@@ -20,3 +38,40 @@ func (s *pgStore) ListUsers(ctx context.Context) (users []User, err error) {
 
 	return
 }
+
+// CreateNewUser = creates a new user in database
+func (s *pgStore) CreateNewUser(ctx context.Context, u User) (newUser User, err error) {
+	tx, err := s.db.Beginx()
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error in beginning user insert transaction")
+		return
+	}
+	_, err = tx.NamedExec(insertUserQuery, u)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error while inserting user into database")
+		return
+	}
+	err = tx.Commit()
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error while commiting transaction inserting user")
+		return
+	}
+	_, newUser, err = s.CheckUserByEmail(ctx, u.Email)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error selecting user from database with email: " + u.Email)
+		return
+	}
+	return
+}
+
+func (s *pgStore) CheckUserByEmail(ctx context.Context, email string) (check bool, user User, err error) {
+	err = s.db.Get(&user, getUserByEmailQuery, email)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
no need to add 
`if err == sql.ErrNoRows ` condition in this func
caller of the func should handle it

Reply:
I have used `if err == sql.ErrNoRows` condition to avoid logging `no row found`  errors and to log only database related errors